### DISCUSSION TITLE
File diffs in pull requests tab eat browser scroll

### DISCUSSION THREAD

**Discussion by smoogipoo** [community/community#8098] [Pull Requests]
Sometimes, when scrolling in the pull requests tab with the cursor positioned over a file diff, the diff area will eat the browser's scroll and scroll itself by 1px or so. It seems to happen more often when you hover over the diff for a bit, as opposed to hovering over it while the page is already scrolling.

The expectation is for the page to scroll, as if the mouse was hovering the sides of the page.

Video of it happening:

https://user-images.githubusercontent.com/1329837/143385490-b01be01b-0776-4eba-8858-f583325f4aaf.mp4

You can see the diff area scrolling by a very small amount when it looks like I'm not doing anything (I'm scrolling as hard as I can).

Browser:
- Firefox 94.0.2

**Comment by michaeljones** [comment]
I realise that +1 comments are not that useful but this is happening to me too on Firefox on Ubuntu and it is driving me a little mad. Feels like my mouse is broken as scrolling just randomly doesn't work but it is only on diff pages and on closer inspection is due to this 1px size diff.

**Reply by smoogipoo** [reply]
I've also submitted a ticket for this and am still waiting for a response at either end.

**Reply by michaeljones** [reply]
I don't know if it can be recommended but from a quick experiment if you follow: https://superuser.com/questions/318912/how-can-i-override-the-css-of-a-site-in-firefox-with-usercontent-css to set up userContext.css stylesheet and then:

```css
@-moz-document domain(github.com) {
    .data.highlight {
        overflow-y: hidden;
    }
}
```
It might help. I don't know the best way of referring to the particular div that is causing the issue. I'm not even sure what the issue is but this helps me on one random diff I loaded up that had the issue. I don't know where to find docs for `@-moz-document` or `domain()` or if there is a way to only target `github.com/*/pull/*/files` or something like that. I also don't know what other things this might impact as I've only tried it for about 2 mins.

**Reply by danuker** [reply]
Thank you for this! It seems to work in FF 97.0.1 as well, improving my sanity.

To reduce the risk of matching unintended elements, I added the `.diff-view` class as a required [ancestor](https://developer.mozilla.org/en-US/docs/Web/CSS/Descendant_combinator):

```css
@-moz-document domain(github.com) {
    .diff-view .data.highlight {
        overflow-y: hidden;
    }
}
```

**Comment by RafalSkolasinski** [comment]
For what is worth in my experience issue was only present on Firefox combined with i3wm. 
On the side note, after creating the `userContent.css` file I had to restart Firefox for it to work. 

Seems to solve the issue so far.

**Reply by danuker** [reply]
I am also using i3 and was affected by this, for what it's worth. But I used i3 since 2018-ish.

**Reply by UnseenBook** [reply]
I assume it's not WM related since I'm on Gnome and also have this issue for a long time now.
It seems related to Firefox since in Chromium I don't experience it. But I would love to see this fixed because it's pretty annoying to have to place the cursor on the side of the page to be able to scroll.
Just tested it without extensions just to be sure and it still happens.

- Firefox: 98.0.2
- Gnome Shell & Mutter: 41.5

**Reply by smoogipoo** [reply]
Yep, I would assume the same - on XFCE here:
- Firefox: 98.0
- XFCE: 4.16

The issue is even worse now since GitHub's previewing the new fullscreen layout with the "Pull Request File Tree" feature...

**Comment by mullr** [comment]
I have this issue as well (Firefox on KDE), and applying this fix using stylus works well. It also fixes scrolling using the arrow keys, and page-up / page-down. 

Posted to https://userstyles.world/style/4652/linux-github-pr-scroll-fix for convenience.

**Comment by apinkney97** [comment]
I can reliably reproduce this (or a similar issue) in Chrome `101.0.4951.64` on Ubuntu 20.04 by clicking the "expand down" button in the bottom left of any file pane until the bottom of the file is reached. When the bottom of the file is reached, a scrollbar appears with about 1 pixel of scroll.

This happens both in unified and split views.

**Reply by mleonhard-flexport** [reply]
I'm seeing this frustrating behavior on GitHub Enterprise with:
* MacBook Pro (16-inch, 2021)
* macOS 12.6 (21G115)
* Chrome Version 105.0.5195.125 (Official Build) (arm64), page zoom=150%.
* Chrome extensions: BlueJeans for Google Calendar, Cleary, Google Docs Offline, LastPass, and Okta.

The scrolling issue occurs every time while using an LG 32GK650F monitor or 2010 Apple Cinema HD Display and a USB mouse.  It occurs less frequently while using the laptop's built-in screen and touchpad, scrolling with 2 fingers.

I've filed a ticket with my company's IT team asking them to file a ticket with Github.
